network.py

```diff
@@ -85,6 +129,10 @@ def accuracy(y_true, y_pred):
 
 def compile(model):
```

Comment:
We might want to do the same here, to allow choosing the optimizer, but we can do it in a follow-up PR.

Reply:
@marco-c changed the compile function a bit more. Hope it looks good. :)